lib/assessment.js

```diff
@@ -8,28 +8,66 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
+<<<<<<< HEAD
+var min = function min(array, callback, initialValue) {
+  var previousValue;
+  if (initialValue !== undefined) {
+    previousValue = initialValue;
+  } else {
+    array = array.slice();
+    previousValue = array.shift();
+  }
+
+  array.forEach(function(currentValue, index, array) {
+    previousValue = callback(previousValue, currentValue, index, array);
+  });
+
+  return previousValue;
+};
+
+var prod = function prod(previousValue, currentValue, index, array) {
+  return previousValue * currentValue;
+=======
 var min = function min(previousValue, currentValue) {
   return previousValue < currentValue ? previousValue : currentValue;
+>>>>>>> origin/solution
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
+<<<<<<< HEAD
+var sum = function sum(numbers) {
+    var total = 0;
+for ( var i = 0; i < numbers.length; i++ ){
+    total += numbers[i];
+}
+    return total;
+}
+// is this what you want????
+=======
 var sum = function sum(previousValue, currentValue) {
   return previousValue + currentValue;
 };
+>>>>>>> origin/solution
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
 var any = function any(array, callback, thisArg) {
+<<<<<<< HEAD
+// No idea how to begin
+// a for loop?
+//I am not understanding callbacks
```

Comment:
This is a great topic to bring up in a one-on-one or talk to with me or Matt after 4:30 today. Callbacks are tricky and we don't expect anyone to understand them intuitively.